### PR TITLE
Set up cargo deny

### PR DESCRIPTION
**Pull request by dkhofer**
Cargo deny is a tool for checking various things about dependencies: 
* Are there any dependencies with known security issues?
* Are there any deprecated dependencies?  (Turns out tempdir was deprecated, I removed it)
* Do all dependencies have a license we like?
* Other cargo file linting

https://github.com/EmbarkStudios/cargo-deny

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -0,0 +1,238 @@
+# This template contains all of the possible sections and their default values
```
Comment (dkhofer) [inline deny.toml:1]: Most of this is default boilerplate

File: deny.toml [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+[licenses]
+# List of explicitly allowed licenses
+# See https://spdx.org/licenses/ for list of possible licenses
+# [possible values: any SPDX 3.11 short identifier (+ optional exception)].
+allow = [
+    "MIT",
+    "Apache-2.0",
+    "Apache-2.0 WITH LLVM-exception",
+    "BSD-3-Clause",
+    "ISC",
```
Comment (dkhofer) [inline deny.toml:96]: https://en.wikipedia.org/wiki/ISC_license

File: deny.toml [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    "Unicode-DFS-2016",
```
Comment (dkhofer) [inline deny.toml:97]: https://spdx.org/licenses/Unicode-DFS-2016.html

File: deny.toml [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+#git-fetch-with-cli = true
+
+# This section is considered when running `cargo deny check licenses`
+# More documentation for the licenses section can be found here:
+# https://embarkstudios.github.io/cargo-deny/checks/licenses/cfg.html
+[licenses]
+# List of explicitly allowed licenses
+# See https://spdx.org/licenses/ for list of possible licenses
+# [possible values: any SPDX 3.11 short identifier (+ optional exception)].
+allow = [
```
Comment (dkhofer) [inline deny.toml:91]: Please make sure these are all acceptable

**Review by Chris7 — APPROVED** [review]
cool tool

**Pull request merged**